### PR TITLE
Source Faker: Add support for PyPi and AirbyteLib entrypoints

### PR DESCRIPTION
**Pull request by aaronsteers**
This adds compatibility with AirbyteLib.

Includes PyPi declaration as defined in:

- https://github.com/airbytehq/airbyte/pull/33529

This PR replaces the related changes in:

- https://github.com/airbytehq/airbyte/pull/33878

This PR also includes a generic Python DevContainer, which can be used for testing in Codespaces. I'm happy to remove it though and move to a different PR if anyone has an objection. I confirmed I was able to run pytest from this decontainer, and that all tests are passing.

### CODE REVIEW

**Comment by aaronsteers** [comment]
I created a generic DevContainer, which can be used for testing in Codespaces. I'm happy to remove it though and move to a different PR if anyone has an objection. I confirmed I was able to run pytest from this decontainer, and that all tests are passing.

**Review by alafanechere — APPROVED** [review]
The `source-faker` changes LGTM.
I raised a couple of concerns / suggestion about the `devcontainer` configuration. 
I'm not using `devcontainer` myself so my suggestion might be unrealistic / not matching what you like to achieve.

Please engage in a more in-depth discussion if you'd like to broaden the adoption of `devcontainer.json` among our connector developers. My main concern is to find a way to guarantee a certain level of consistency between the dev environment (managed by `devcontainers.json`) and the build environment (managed by `airbyte-ci`) so that developer do not face late CI failures / bad surprises on release.

**Review by flash1293 — APPROVED** [review]
Approving for faker changes, @alafanechere s comments on dev container make sense to me

File: .devcontainer/python-connectors-generic/devcontainer.json
[the remark this answers is not in the document]
Comment (aaronsteers) [reply .devcontainer/python-connectors-generic/devcontainer.json:1]: @alafanechere - I tested successfully and was able to run `airbyte-ci` without issue, thanks to the `make` option.

Happy to keep iterating on this. For now, it gives a baseline experience that's much better than starting with a blank of generic container.

File: .devcontainer/python-connectors-generic/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        "git.openRepositoryInParentFolders": "always"
+      }
+    }
+  },
+
+  // Mark the root directory as 'safe' for git.
+  "initializeCommand": "git config --add safe.directory /workspaces/airbyte",
+
+  // Setup airbyte-ci on the container:
+  "postCreateCommand": "make tools.airbyte-ci-dev.install",
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply .devcontainer/python-connectors-generic/devcontainer.json:53]: It works fine as-is for now. In the cloud, I think docker-in-docker is correct, and probably the best we can get. There's also a 'docker-from-docker' option that might work better locally - but I don't have familiar with that as of yet, and so I'd like to keep as is for now.

> Please engage in a more in-depth discussion if you'd like to broaden the adoption of devcontainer.json among our connector developers.

Agreed. No need to push this as of now. I think over time we will get to a good place where we have collective confidence that the containers give a solid and consistent experience.

**Pull request merged**